app.py

```diff
@@ -124,47 +124,31 @@ def launch_publish(self, entity_type, entity_ids):
                 if not v.get("published_files"):
                     self.log_error("Sorry, this can only be used on Versions with an associated Published File.")
                     return
-                publish_id = v["published_files"][0]["id"]
+                if len(v["published_files"]) == 1:
+                    publish_id = v["published_files"][0]["id"]
+                # if there are multiple published files, pick one.
+                else:
+                    try:
+                        selected = self.execute_hook(
+                            "hook_multiple_publishes",
+                            published_files=v["published_files"],
+                        )
+                        publish_id = selected["id"]
+                    except TankError, e:
+                        self.log_error("Failed to select a published file from multiple: %s" % e)
+                        return
             else:# == "TankPublishedFile":
                 v = self.shotgun.find_one("Version", [["id", "is", entity_ids[0]]], ["tank_published_file"])
                 if not v.get("tank_published_file"):
                     self.log_error("Sorry, this can only be used on Versions with an associated Published File.")
                     return
                 publish_id = v["tank_published_file"]["id"]
-            
         else:
             publish_id = entity_ids[0]
 
         # first get the path to the file on the local platform
         d = self.shotgun.find_one(published_file_entity_type, [["id", "is", publish_id]], ["path", "task", "entity"])
```

Comment:
Here we're querying again SG, but we already did that in the hook, why not re-using `selected` here?